components/org.wso2.carbon.identity.sp.metadata.saml2/src/main/java/org/wso2/carbon/identity/sp/metadata/saml2/util/Parser.java

```diff
@@ -198,8 +198,21 @@ public SAMLSSOServiceProviderDO parse(String metadata, SAMLSSOServiceProviderDO
             List<RoleDescriptor> roleDescriptors = entityDescriptor.getRoleDescriptors();
             //TODO: handle when multiple role descriptors are available
             //assuming only one SPSSO is inside the entitydescripter
-            RoleDescriptor roleDescriptor = roleDescriptors.get(0);
-            SPSSODescriptor spssoDescriptor = (SPSSODescriptor) roleDescriptor;
+            RoleDescriptor roleDescriptor;
+            SPSSODescriptor spssoDescriptor;
+
+            if (roleDescriptors.size() > 0) {
+                roleDescriptor = roleDescriptors.get(0);
+            } else {
+                throw new InvalidMetadataException("Role descriptor not found.");
+            }
+
+            if (roleDescriptor instanceof SPSSODescriptor) {
+                spssoDescriptor = (SPSSODescriptor) roleDescriptor;
+            } else {
+                throw new InvalidMetadataException("Invalid role descriptor class found.");
+            }
```

Comment:
```suggestion
            if (!roleDescriptor instanceof SPSSODescriptor) {
                throw new InvalidMetadataException("Invalid role descriptor class found.");
            } 
            spssoDescriptor = (SPSSODescriptor) roleDescriptor;
```

We can get rid of else blocks by using this pattern right? Use it in other places as well

Reply:
Fixed.